src/main/java/com/iemr/hwc/fhir/provider/patient/PatientExtProvider.java

```diff
@@ -32,4 +44,70 @@ public MethodOutcome createPatient(HttpServletRequest theRequest, @ResourceParam
+                EncounterExt encounter = new EncounterExt();
+                encounter.setId(1L+"");
+                encounter.setProviderServiceMapId(new StringType(benDetails.getBenVisitDetails().getProviderServiceMapID()+""));
+                encounter.setVanID(new StringType(benDetails.getBenVisitDetails().getVanID()+""));
+                encounter.setParkingPlaceID(new StringType(benDetails.getBenVisitDetails().getParkingPlaceID()+""));
+                encounter.setCreatedBy(new StringType(benDetails.getBenVisitDetails().getCreatedBy()));
+                encounter.setSessionID(new StringType(benDetails.getSessionID()+""));
+                encounter.setBeneficiaryID(new StringType(benDetails.getBeneficiaryID()+""));
+                encounter.setBenFlowID(new StringType(benDetails.getBenFlowID()+""));
+                encounter.setBeneficiaryRegID(new StringType(benDetails.getBenVisitDetails().getBeneficiaryRegID()+""));
+
+                String status = benDetails.getBenVisitDetails().getVisitFlowStatusFlag();
+                if(status !=null && status.trim().equals("N")){
+                    encounter.setStatus(Encounter.EncounterStatus.NULL);
+                }
+
+                List<CodeableConcept> listConcept = new ArrayList<>();
+                CodeableConcept concept2 = new CodeableConcept();
+                List<Coding> listCoding2 = new ArrayList<>();
+                Coding coding2 = new Coding();
+                coding2.setSystem("http://snomed.info/sct");
+                coding2.setDisplay(benDetails.getBenVisitDetails().getVisitCategory());
+                listCoding2.add(coding2);
+                concept2.setCoding(listCoding2);
+                listConcept.add(concept2);
+                encounter.setType(listConcept);
+
+                CodeableConcept concept = new CodeableConcept();
+                List<Coding> listCoding = new ArrayList<>();
+                Coding coding = new Coding();
+                coding.setSystem("http://snomed.info/sct");
+//            coding.setDisplay(benDetails.getVisitCategory());
+                listCoding.add(coding);
+                concept.setCoding(listCoding);
+                encounter.setServiceType(concept);
+
+                Coding coding3 = new Coding();
+                coding3.setSystem("http://terminology.hl7.org/CodeSystem/v3-ActCode");
+                coding3.setCode("");
+                coding3.setDisplay("");
+                encounter.setClass_(coding3); // Need clarifications
+
+                List<CodeableConcept> listConcept2 = new ArrayList<>();
+                CodeableConcept concept3 = new CodeableConcept();
+                concept3.setText(benDetails.getBenVisitDetails().getVisitReason());
+                listConcept2.add(concept3);
+                encounter.setReasonCode(listConcept2);
+
+                listRes.add(encounter);
+            }
+        }
+        catch (Exception e){
+            e.printStackTrace();
```

Comment:
Give error in the api response incase of exception

Reply:
An exception is thrown automatically if an error occurred.
Or you want an explicit exception response ?

Reply:
You are catching the exception. Will it be thrown automatically?

Reply:
Yes it will be thrown automatically. I just want to print the error on log file that's why I use printStackTrace() function